src/x-labels.js

```diff
@@ -23,7 +24,10 @@ export default class XLabels extends React.Component {
     }
 
     render() {
-        const format = 'month' === this.props.display ? 'dddd' : 'ddd, MMM Do';
```

Comment:
I had forked this repo and clearly have encountered similar problems as you. What do you think about this solution:
```javascript
export default class XLabels extends React.Component {
  ...

  get dateFormat() {
        const defaultFormat = 'month' === this.display ? 'dddd' : 'ddd, MMM Do';
        return this.props.dateFormat || defaultFormat;
    }

  ...

  render() {
        return (
            <div className="x-labels">{this.days.map(day => <div key={day.format('YYYYMMDD')} className="day-label">
                {day.format(this.dateFormat)}
            </div>)}
            </div>
        );
    }
```

Reply:
yes, I think adding a getter would be preferable, 👍 to something like outlined by @rkorzeniec. The proposed change below has a small bug where the provided dateFormat wouldn't be used if `display === 'month'`

Reply:
I've add the getter. Please check my new changes.